Windows Kernel Build Failure on AZP

### Describe the bug

Without any change on our side, AZP is now failing to build Windows kernel mode. There was likely a new DDK installed and it's adding additional validation rules:

https://dev.azure.com/ms/msquic/_build/results?buildId=123219&view=logs&j=7833c698-5b4c-5fa1-a964-7b79e00bc606&t=ade1039f-c905-56de-0332-d7fac3c4e8d6

```
##[error]src\core\library.h(324,0): Error C33010: Unchecked lower bound for enum Type used as index.
```

--- a/src/core/library.h
+++ b/src/core/library.h
@@ -318,7 +318,7 @@ QuicPerfCounterAdd(
     _In_ int64_t Value
     )
 {
-    QUIC_DBG_ASSERT(Type < QUIC_PERF_COUNTER_MAX);
+    QUIC_DBG_ASSERT(Type >= 0 && Type < QUIC_PERF_COUNTER_MAX);
     uint32_t ProcIndex = QuicProcCurrentNumber();
     QUIC_DBG_ASSERT(ProcIndex < (uint32_t)MsQuicLib.PartitionCount);
     InterlockedExchangeAdd64(&(MsQuicLib.PerProc[ProcIndex].PerfCounters[Type]), Value);

